Add LM Studio support

pwilkin opened the pull request:
Added LM Studio support, separately from Ollama support so both of those can be utilized independently.

pwilkin committed:
Updated for linux dev run, changed publish repo, minor audit fixes

pwilkin committed:
Remove Mac-related scripts

pwilkin committed:
Added LM Studio support

pwilkin committed:
It's usually a good idea to add the backend too...

pwilkin committed:
Revert change to config files for merge

pwilkin committed:
Minor bug.

pwilkin committed:
Added details to package.json, bumped version number for Linux LM Studio support build

pwilkin committed:
Revert "Added details to package.json, bumped version number for Linux LM Studio support build"

This reverts commit a221573ef6ef309e8a40b7b91967975917b56552.

wwwillchen commented:
Thanks for the PR.

Instead of adding the @lmstudio/sdk dependency, I think we can just rely on the OpenAI-compatible API - I'd like to avoid dependencies when reasonable to keep the download size as manageable as possible.

We should be able to fetch the LM Studio models using the OpenAI compatible that LM studio provides: https://lmstudio.ai/docs/app/api/endpoints/openai

I see that you're using the OpenAI client to do the model call, but you can actually use: https://sdk.vercel.ai/providers/openai-compatible-providers/lmstudio.

pwilkin commented:
> Instead of adding the @lmstudio/sdk dependency, I think we can just rely on the OpenAI-compatible API - I'd like to avoid dependencies when reasonable to keep the download size as manageable as possible.
> We should be able to fetch the LM Studio models using the OpenAI compatible that LM studio provides: https://lmstudio.ai/docs/app/api/endpoints/openai

There's a problem with that, though - the OpenAI-compatible `/v1/models` path lists *loaded* models, not available ones. So we'd have to manually fetch from the LM Studio-specific endpoint (not that it's a big problem).

> I see that you're using the OpenAI client to do the model call, but you can actually use: https://sdk.vercel.ai/providers/openai-compatible-providers/lmstudio.

You're right :)

Okay, I'll try to remove the LMStudio SDK (it's not very well documented anyways) and just do a manual fetch from the models endpoint to get the models.

pwilkin commented:
Alright, I removed the references and added the manual fetch. While I was at it, I also fixed all linter errors so that npm run lint returns 0 errors :)

wwwillchen commented:
Let's keep this PR tightly focused to LM studio only changes - I appreciate the lint fixes, but it's harder to review.

pwilkin commented:
> Let's keep this PR tightly focused to LM studio only changes - I appreciate the lint fixes, but it's harder to review.

Aight, will redo.

pwilkin committed:
LM Studio changes only

pwilkin commented:
Sorry, haven't had time for a few days, but I cleaned up the PR and rebased on current main branch.

wwwillchen commented on package.json:
I think all the changes in this file should be reverted except for @ai-sdk/openai-compatible

pwilkin replied:
Done.

wwwillchen commented on package-lock.json:
see below  - revert

pwilkin replied:
Done.

wwwillchen commented on .github/workflows/release.yml:
revert

pwilkin replied:
Done.

wwwillchen commented on src/components/ModelPicker.tsx:
```diff
@@ -63,21 +82,24 @@ export function ModelPicker({
   return (
-    <DropdownMenu open={open} onOpenChange={setOpen}>
+    <DropdownMenu open={open} onOpenChange={(newOpen) => {
+      console.log("DropdownMenu onOpenChange:", newOpen);
```
remove

wwwillchen commented on src/ipc/handlers/local_model_ollama_handler.ts:
remove all console logs - use logger if something needs to be logged

pwilkin replied:
Done.

wwwillchen reviewed (changes requested):
Overall changes look good but need some code clean-ups and reverting earlier changes before we can merge this.

UX-wise, I think Ollama and LM Studio models should be grouped into one category "Local models" - they can still be in their own sub-category (e.g sub-sub-category), but I can take care of this after this PR if you want.

pwilkin committed:
Cleanup for merge

pwilkin commented:
Aight, made the changes, as for the button logic, I'd leave it to you - I'm *terrible* at layouting :>

wwwillchen reviewed (approved):
Thanks! Merging
